labs/a5.md

```diff
@@ -0,0 +1,336 @@
+
+* `domain`
+Using this option will specific a local domain name.  Short queries, which are queries that don't contain any domain identifiers, then have the local domain appended to them during DNS queries.
+
+  To understand this better take `death` as one of the machines within the OCF domain, `ocf.berkeley.edu`.  One can issue a DNS query for death by typing `dig death.ocf.berkeley.edu` but that's an awful lot to type.  By specifying `domain ocf.berkeley.edu` in **/etc/resolv.conf** the query can be shortened to just `dig death`.  In fact, any tool that takes a domain name can now use this shortened version, i.e. `ping death`.  This is because your machine's resolver is responsible for translating this domain name, and the `domain` configuration automatically appends the written domain to these short queries.
+* `search`
+The format for this option is `search <search-list>`.  Using this options specifies a list of domain names to iterate through when attempting to look up queries.  
+
+  Let's examine an example use case, imagine we owned two networks `ocf.berkeley.edu` and `xcf.berkeley.edu` and wanted to query a machine which may either be in either network.  To enable this we can simply add the line `search ocf.berkeley.edu xcf.berkeley.edu`.  Queries to resolve a domain name will now append those listed domains in order until a successful DNS response.  If we assume `death` is on `ocf.berkeley.edu` and another machine, `life`, is on `xcf.berkeley.edu`, both `dig death` and `dig life` are now resolved properly thanks to our configuration.
+
+One thing to note is that **`search` and `domain` are mutually exclusive keywords** and having both defined causes the last instance to take precedence and override earlier entries.
+
+* `nameserver`
+The `nameserver` keyword is fairly self explanatory and follows the format of `nameserver <ip-address>` where `<ip-address>` is the IP address of the intended name server.  One can have up to `MAXNS`, defined in `<resolv.h>`,  `nameserver` entries in this file.  The resolver will query nameservers in the same order as they are written in the file.
+
+Following are additional useful configurable options in this file. Options are defined in this format `options <option1> [additional-options]`.  Some example options follow below:
+
+* `ndots`
+This option is in the format `ndots:n` configures the threshold,`n`, at which  an initial absolute query is made.  Since the default value for this option is 1, any name with at least 1 dot will first be queried as an absolute name before appending domains from `search`.  When less than `ndots` are present, the queries automatically begin appending elements in `<search-list>`.
```

Comment:
I think this will be a bit confusing, I mean I had to read it like 3 times to understand what the option does. Maybe give an example or remove it if it isn't that important?

Reply:
example provided